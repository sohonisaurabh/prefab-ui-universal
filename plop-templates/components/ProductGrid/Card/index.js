/** @jsx jsx */
import React from "react";
import AddToCartAction from "../AddToCartAction/index";
import { jsx, css } from "@emotion/core";

const cardStyle = css`
  max-width: 400px;
  background: #fff;
  padding: 0.25rem;
  margin-top: 1rem;
  border-radius: 4px;
`;

const ulStyle = css`
  list-style-type: none;
  padding: 0;
  margin-top: 0;
  display: flex;
  justify-content: space-between;
`;

const liStyle = css`
  list-style: none;
  padding: 0.3em 0;
  border-radius: 1em;
`;

const imageStyle = css`
  height: 350px;
`;

const nameStyle = css`
  margin-bottom: 0;
`;

const Card = props => {
  const { product, store } = props;
  return (
    <div css={cardStyle}>
      <img css={imageStyle} src={product.picUrl} />
      <h4 css={nameStyle}>{product.productName}</h4>
      <aside>
        <ul css={ulStyle}>
          <li css={liStyle}>Price: Rs {product.price}</li>
        </ul>
        <AddToCartAction store={store} item={product} />
      </aside>
    </div>
  );
};

export default Card;
