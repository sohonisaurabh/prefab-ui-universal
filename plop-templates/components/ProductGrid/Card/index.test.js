import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';

it('Card renders correctly', () => {
  const product = {
    productName: 'abc',
    picUrl: 'http://skadl/.png',
    price: 1500,
  };
  const tree = renderer.create(<Card product={product} />).toJSON();
  expect(tree).toMatchSnapshot();
});
