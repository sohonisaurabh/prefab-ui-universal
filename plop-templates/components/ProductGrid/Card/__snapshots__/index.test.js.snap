// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="css-1p8gbkt-Card"
>
  <img
    className="css-q0fy04-Card"
    src="http://skadl/.png"
  />
  <h4
    className="css-1wg6z3u-Card"
  >
    abc
  </h4>
  <aside>
    <ul
      className="css-11juu7-Card"
    >
      <li
        className="css-1kiw5wj-Card"
      >
        Price: Rs 
        1500
      </li>
    </ul>
    <button
      className="css-1cpoxhp"
      onClick={[Function]}
    >
      Add to Cart
       
    </button>
  </aside>
</div>
`;
